src/components/search/search.component.tsx

```diff
@@ -92,31 +148,32 @@ const Search = ({
     inputRef,
   ]);
 
-  const handleChange = (e) => {
+  let buttonProps = {};
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
     if (onChange) {
-      onChange(e);
+      onChange(event);
     }
 
     if (!isControlled) {
-      setSearchValue(e.target.value);
+      setSearchValue(event.target.value);
     }
   };
 
-  const handleFocus = (e) => {
+  const handleFocus = (event: React.FocusEvent<HTMLInputElement>) => {
     setIsFocused(true);
     if (onFocus) {
-      onFocus(e);
+      onFocus(event);
     }
   };
 
-  let buttonProps = {};
   if (searchButton && onClick) {
     buttonProps = {
-      onClick: (ev) => {
+      onClick: () => {
```

Comment:
This seems to be a change in behaviour, at least in theory. The old version called the `onClick` prop with the `name` and `id` from the actual event target, whereas this new version will always provide the `name` and `id` from the props provided to `Search`.  I don't think these will be the same because the event target will be the search button itself (or possibly a child of it), not the actual Search component.

So I don't think this works, and even if it does, we shouldn't be changing behaviour in a TS refactor.

Please let me know if I'm misunderstanding something! :)

Reply:
This refactor is not changing anything, as the event could only originate in the search button, or button icon and none of them have either `id` or `name` attribute, so value of these were always defaulting to those of the component.

Reply:
I see now - thanks for clarifying this.